notebooks/a01_manifold_class.ipynb

```diff
@@ -0,0 +1,1630 @@
+{
```

Comment:
<p>The methods of <code>LevelSet</code> are similar to the methods of <code>OpenSet</code>	, but their implementation is different.</p><p> </p><p>For example, observe the implementation of the <code>belongs</code> methods. For a general level set, in order to verify if a point belongs to the level set, we should verify that the constraint is met which is done with the line:</p><p><code>constraint = gs.isclose(self.submersion(point), value, atol=atol)</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/geomstats/geomstats/pull/1340/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='61'/>

Reply:
<p>I changed it to:</p><p> </p><p>You can run the code below to see the contents of the <code>LevelSet</code> class. The methods of LevelSet are similar to the methods of OpenSet , but their implementation is different. Recall from the (Local Implicit Function) definition, which <code>LevelSet</code> is an implementation of, that a manifold of this type must satisfy:</p><p> </p><p>"For every $p \in M$, there exists an open set $U \in \mathbb{R}^{N}$ and a smooth map $f: U \to \mathbb{R}^{N-d}$ that is a submersion at p, such that $U \cap M = f^{-1}$({0})."</p><p> </p><p><code>OpenSet</code> is an implementation of the first manifold definition (Local Parametrization), and therefore need not follow the (Local Implicit Function) rule above. This is the reason that, <code>OpenSet</code> methods and <code>LevelSet</code> methods are implemented differently. As an example of these different implementations: observe the implementation of the belongs methods in <code>LevelSet</code>. For a general level set, in order to verify if a point belongs to the level set, we should verify that the (Local Implicit Function) definition constraint is met, which is done with the line</p><p> </p><p><code>constraint = gs.isclose(self.submersion(point), value, atol=atol)</code></p>